RULEAPI-635: Document testing process (#695)
SonarSource/rspec: docs/testing.adoc

= Testing changes

> This document explains how to test changes to the frontend, CI scripts, etc...

Author: Why not indeed. I'll make the change, but two questions:
 1. The difference between "CI configuration" and "CI scripts" here is thin -- the config is some kind of script -- and may cause more confusion. So I wonder if it wouldn't be better to just say "CI". WDYT?
 2. Should I use "RSPEC tools" below too instead of "rspec-tools", or vice-versa? (I'm not sure what's the "official" spelling.)

> This document explains how to test changes to the frontend, CI scripts, etc...

Reviewer: 1. Yes, I think just "CI" should work.
2. We are making the "official" spelling up :) I think rspec-tools should be used in both places - it is the name of the python application

> This document explains how to test changes to the frontend, CI scripts, etc...

Author: Perfect then :-)

```suggestion
This document explains how to test changes to the frontend, CI, and rspec-tools.
```

Refer to the <<../README.adoc#AddModifyRule,main documentation>> if you want to modify rules.

== Testing the frontend

If your changes are small and noncontroversial, you can directly create a branch in this repository.

However, when you need to show reviewers how the website will look with your changes,
you can use a fork with your own GitHub Pages.
See <<forking>> for more details.

To test your changes locally, you can start a local HTTP server, such as ``npm start``.
Refer to the frontend documentation for more details.

== Testing the rspec-tools

Modifications to the rspec-tools do not require any special process.
You can therefore use a regular PR from a branch in this repository.

== Testing GitHub Actions

There are two strategies to test GitHub Actions: either use your own branch in this repository, or create a fork.

Choose the solution your need based on the impact your changes can have.
For example, if the changes may spam/modify other Pull Requests, it is wiser to use your own fork.

> With the former, GitHub will pick the workflow definitions from your branch whether its in a fork or not.

Author: Oups, I moved the sentence and forgot to update this part.

```suggestion
Note that GitHub will pick the workflow definitions from your branch whether it is in a fork or not.
```
should be better.

> With the former, GitHub will pick the workflow definitions from your branch whether its in a fork or not.

Reviewer: LGTM
